build.js

```diff
@@ -0,0 +1,423 @@
+const path = require('path')
+const Hyperschema = require('hyperschema')
+
+const DIR = path.join(__dirname, 'encoding')
+const SPEC = path.join(DIR, 'spec/autobase')
+
+const schema = Hyperschema.from(SPEC, { versioned: true })
+const autobase = schema.namespace('autobase')
+
+autobase.require(path.join(DIR, 'legacy.js'))
+
+autobase.register({
+  name: 'checkout',
+  compact: true,
+  fields: [
+    {
+      name: 'key',
+      type: 'fixed32',
+      required: true
+    },
+    {
+      name: 'length',
+      type: 'uint',
+      required: true
+    }
+  ]
+})
+
+autobase.register({
+  name: 'clock',
+  array: true,
+  type: '@autobase/checkout'
```

Comment:
i cant remember if we have any tests where this isnst declared compact here but the type is, might be safer to just add compact: true here also, for saving our ass in case thats buggy